convert2rhel/actions/post_conversion/hostmetering.py

```diff
@@ -0,0 +1,238 @@
+# Copyright(C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+
+import logging
+import os
+
+from convert2rhel import actions, systeminfo
+from convert2rhel.pkgmanager import call_yum_cmd
+from convert2rhel.subscription import get_rhsm_facts
+from convert2rhel.systeminfo import system_info
+from convert2rhel.utils import run_subprocess
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigureHostMetering(actions.Action):
+    """Configure host metering on a machine if it's needed.
+
+    env_var: str
+        Content of CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable.
+    """
+
+    id = "CONFIGURE_HOST_METERING_IF_NEEDED"
+    env_var = None
+
+    def run(self):
+        """
+        Decide whether to install, enable and start host-metering on the system based on the
+        CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable.
+
+        Behavior can be controlled CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable:
+        - "auto": host-metering will be configured based on the above conditions
+        - env var not present: host-metering will not be configured
+        - "force": forces configuration of host-metering (e.g., even if not running on a hyperscaler)
+        - any other value: Will be ignored and host metering will not be configured.
+
+        :return: True if host-metering is configured successfully, False otherwise (used mainly for testing).
+        :rtype: bool
+        """
+        logger.task("Final: Configure host-metering")
+
+        super(ConfigureHostMetering, self).run()
+
+        self.env_var = os.environ.get("CONVERT2RHEL_CONFIGURE_HOST_METERING", None)
+        if not self._check_env_var():
+            return False
+
+        if system_info.version.major != 7 and self.env_var == "auto":
```

Comment:
This is a bit different from the original code, is it intended? https://github.com/oamg/convert2rhel/blob/main/convert2rhel/hostmetering.py#L73

Reply:
After good point from @Venefilyn here: https://github.com/oamg/convert2rhel/pull/1311#discussion_r1688132019 I spent some time on refactoring the conditions a bit. I think having the non negation state of the `env_var` is easier to read. And since it can have only `auto` or `force` (we check it in the code before), I believe it's fine, and the functionality is the same.